Jenkinsfile

```diff
@@ -17,7 +17,10 @@
 
 pipeline {
     agent {
-        label 'master'
+        node {
+            label 'master'
+            customWorkspace "workspace/${env.BUILD_TAG}"
```

Comment:
cc @RyanLassigBanks @rberg2 on this bit ^^. Solves an issue with the workspace name starting with `-`

Reply:
Actually, it looks like we may want to instead pass in `-Djenkins.branch.WorkspaceLocatorImpl.PATH_MAX=0` to Jenkins to just stop the path truncation behavior:

https://issues.jenkins-ci.org/browse/JENKINS-38706?focusedCommentId=274531&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-274531

Reply:
@RyanLassigBanks says we should use the method you've got here instead of passing in the flag at startup